File: Data Structures & Algorithms/merge-triplets-to-form-target/submission-2.py
class Solution:
    def mergeTriplets(self, triplets: List[List[int]], target: List[int]) -> bool:
        x = y = z = False
        for a, b, c in triplets:
            if a > target[0] or b > target[1] or c > target[2]:
                continue
            x |= a == target[0]
            y |= b == target[1]
            z |= c == target[2]
            if x and y and z:
                return True
        
        return False

File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> diff;
        int index = 0;
        for (int num : nums) {
            int cur_diff = target - num;
            auto it = diff.find(cur_diff);

            if (it != diff.end()) {
                return {it->second, index};
            }

            diff[num] = index++;
        }
    }
};
